listenbrainz/webserver/static/js/src/follow/UserListModalEntry.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import FollowButton from "../FollowButton";
+import SimilarityScore from "../SimilarityScore";
+
+export type UserListModalEntryProps = {
+  mode: "follow-following" | "similar-users";
+  user: ListenBrainzUser;
+  loggedInUser: ListenBrainzUser | null;
+  loggedInUserFollowsUser: boolean;
+  similarityScore: number;
+  updateFollowingList: (
+    user: ListenBrainzUser,
+    action: "follow" | "unfollow"
+  ) => void;
+};
+
+const UserListModalEntry = (props: UserListModalEntryProps) => {
+  const {
+    mode,
+    user,
+    loggedInUserFollowsUser,
+    loggedInUser,
+    similarityScore,
+    updateFollowingList,
+  } = props;
+  return (
+    <>
+      <div key={user.name}>
+        <div>
+          <a
+            href={`/user/${user.name}`}
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {user.name}
+          </a>
+          {loggedInUser && mode === "similar-users" && (
+            <SimilarityScore
+              similarityScore={similarityScore}
+              user={user}
+              type="compact"
+            />
+          )}
+        </div>
+        {loggedInUser && (
+          <FollowButton
+            type="block"
+            user={{ name: user.name }}
```

Comment:
Similarly, I don't understand why you're creating a new object with just the name rather than pass `user={user}`